core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -234,9 +289,13 @@ object FiberRefs {
      */
     def diff(oldValue: FiberRefs, newValue: FiberRefs): Patch = {
```

Comment:
Suggestion: I think the code of this function would be a bit more readable this way:

```scala
def diff(oldValue: FiberRefs, newValue: FiberRefs): Patch = {
      val (removed, patch) =
        newValue.fiberRefLocals.foldLeft[(FiberRefs, Patch)](oldValue -> empty) {
          case ((fiberRefs, patch), (fiberRef, Value(StackEntry(_, newValue, _) :: _, _))) =>
            fiberRefs.getOrNull(fiberRef) match {
              case null =>
                fiberRefs -> patch.combine(Add(fiberRef, newValue.asInstanceOf[fiberRef.Value]))
              case oldValue =>
                val newPatch =
                  if (oldValue == newValue) patch
                  else
                    patch.combine(
                      Update(
                        fiberRef.asInstanceOf[FiberRef.WithPatch[fiberRef.Value, fiberRef.Patch]],
                        fiberRef.diff(oldValue.asInstanceOf[fiberRef.Value], newValue.asInstanceOf[fiberRef.Value])
                      )
                    )

                fiberRefs.delete(fiberRef) -> newPatch
            }
        }

      removed.fiberRefLocals.foldLeft(patch) { case (patch, (fiberRef, _)) => patch.combine(Remove(fiberRef)) }
    }
```